src/checkpoint/checkpoint_api.h

```diff
@@ -51,29 +51,93 @@
 
 namespace checkpoint {
 
+/// Callback for user to allocate bytes during serialization
 using BufferCallbackType = std::function<char*(std::size_t size)>;
 
+/// Return of serialize that contains the buffer and size serialized
 struct SerializedInfo {
   virtual std::size_t getSize() const = 0;
   virtual char* getBuffer() const = 0;
   virtual ~SerializedInfo() { }
 };
 
-using SerializedInfoPtrType = std::unique_ptr<SerializedInfo>;
-using SerializedReturnType = SerializedInfoPtrType;
+/// Convenience typedef for \c std::unique_ptr<SerializedInfo>
+using SerializedReturnType = std::unique_ptr<SerializedInfo>;
 
+/**
+ * \brief Serialize \c T into a byte buffer
+ *
+ * Serializes an object recursively by invoking the \c serialize
+ * functions/methods recursively.
+ *
+ * \param[in] target the \c T to serialie
+ * \param[in] fn (optional) callback to supply buffer for to allow user
+ * allocation of the produced byte buffer. The callback will pass the number of
+ * bytes required and return a char* will at least that many bytes.
```

Comment:
Typo 'will' -> 'with'

The text should be clarified to read

" The callback will be passed the number of bytes required and return a char* to a buffer of at least that many bytes."

Reply:
Fixed